cmd/create/idp/htpasswd.go

```diff
@@ -134,6 +147,15 @@ func buildUserList(cmd *cobra.Command, r *rosa.Runtime) *cmv1.HTPasswdUserListBu
 			}
 			userList[u] = p
 		}
+	} else if usersfile != "" {
+		err := parseHtpasswordFile(&userList, usersfile)
+		if err != nil {
+			r.Reporter.Errorf(
+				"Failed to load Htpasswd file '%s': %v", usersfile, err)
+			os.Exit(1)
+		}
+		//password in htpasswd are already and do not need to be hashed again in CS
+		hashed = true
```

Comment:
Is it always expected that a file input will be pre hashed? Is there any way we can check this somehow :thinking:

Reply:
@gdbranco : if you  look at the htpsswd utility used to build htpasswd file, by default they are hashed ( you can ofcourse  force change the type/status of encryption )

This concept is the same  in OCP as well.  For Htpasswd IDP, OCP just lets you upload any well-formed htpasswd file and I think the idea here  is to be able to achieve the same thing with  ROSA.

While doing the design review on this,  I did bring up with @vkareh the fact that we may not be able to validate the passwords like we do with the plain-text ones.
But everybody seemed ok with it, we want to give the users the ability to upload htpasswd file and let them choose the encryption type and status.